components/CreateEventsWizard.tsx

```diff
@@ -0,0 +1,353 @@
+  const handleRepeatStates = (
+    repeat_param: boolean,
+    interval_param: number,
+    weekDays_param: number[],
+    endType_param: string,
+    endDate_param: Date,
+    count_param: number
+  ): void => {
+    setRepeat(repeat_param);
+    if (repeat_param) {
+      setInterval(interval_param);
+      setWeekDays(weekDays_param);
+      setEndType(endType_param);
+      setEndDate(endDate_param);
+      setCount(count_param);
+
+      // rrule text to display on button
+      const rule = new RRule({
+        interval: interval_param,
+        freq: RRule.WEEKLY,
+        byweekday: weekDays_param,
+      });
+      setRruleText(rule.toText());
+    } else {
+      setRruleText("Never Repeat");
+    }
+  };
+
+  // handles create wizard submit
+  const handleSubmit = async (): Promise<void> => {
+    setErrMsg("");
+    setLoading(true);
+    let lang = "unknown";
+    if (name.toLowerCase().includes("java")) {
+      lang = "Java";
+    } else if (name.toLowerCase().includes("python")) {
+      lang = "Python";
+    }
+
+    // construct rrule object
+    let rrule;
+    if (repeat) {
+      if (endType === "on") {
+        rrule = new RRule({
+          dtstart: startDate,
+          interval: interval,
+          freq: RRule.WEEKLY,
+          byweekday: weekDays,
+          until: endDate,
+        });
+      } else if (endType === "after") {
+        rrule = new RRule({
+          dtstart: startDate,
+          interval: interval,
+          freq: RRule.WEEKLY,
+          byweekday: weekDays,
+          count: count,
+        });
+      } else {
+        rrule = new RRule({
+          dtstart: startDate,
+          interval: interval,
+          freq: RRule.WEEKLY,
+          byweekday: weekDays,
+        });
+      }
+    } else {
+      rrule = new RRule({
+        dtstart: startDate,
+        count: 1,
+      });
+    }
+    const rruleStr = rrule.toString();
+
+    const createEvent: CreateClassEvent = {
+      name: name,
+      startTime: startTime,
+      endTime: endTime,
+      timeZone: DateTime.local().zoneName,
+      rrule: rruleStr,
+      language: lang,
+      neverEnding: repeat && endType === "never",
+      backgroundColor: colorMap[color],
+      teachers: teachers.split(","),
+    };
+
+    try {
+      // Create class event and calendar information
+      const classEvent = await client.createClassEvent(createEvent);
+      try {
+        // create class from event id
+        const createClass: CreateClass = {
+          eventInformationId: classEvent.eventInformationId,
+          minLevel: minLevel,
+          maxLevel: multipleLevels ? maxLevel : minLevel,
+          rrstring: rruleStr,
+          language: lang,
+          startTime: classEvent.startTime,
+          endTime: classEvent.endTime,
+        };
+        await client.createClass(createClass);
+      } catch (err) {
+        setErrMsg("Error on class creation");
+        setLoading(false);
+        return;
```

Comment:
Since the backend has some functionality to detect the specific error the backend is sending. Something like this should do the trick:
```javascript
catch (err) {
      if (axios.isAxiosError(err) && err.response) 
        setErrMsg(err.response.data);
      else if (err instanceof Error)
        setErrMsg(err.message);
      else setErrMsg("Error")
      setLoading(false);
      return;
```